### PR TITLE
feature-hint

### LINKED ISSUE #1283: [Enhancement] Drawer for feature-hint
**Is your enhancement request related to a problem? Please describe.**
Some features are kind of hidden in the UI, e.g. cell swipe actions, features in the overflow menu or triggered after rotation of the device.
User are not always aware of the ‘hidden’ features and needs onboarding to them i context.

**Describe the solution you'd like**
Show a feature onboarding hint (drawer) when relevant and in context.
The timing is essential and would be triggered by different rules in context.
Only one feature onboarding hint can be triggered at a time.
The feature onboarding hint present the feature and how to use/activate it, e.g. a specific gestures.

When triggered, the onboarding hint shows:

1. drawer moves in from the bottom of the page
2. the drawer ist fixed to the bottom
3. the content of the drawer defines it’s height (probably the same min-height as the drawer)
4. the drawer width is 100% of the device screen, but max. 428px on bigger screens (tablet/web)
5. the drawer contains af title, a close button, an optional svg-image/svg-animation and a text area
6. the drawer background is grey-6 (dark)
7. no backdrop behind the drawer

Normally a drawer is modal with backdrop. But in this circumstance the drawer is without blocking backdrop, so it still is possible to interact with the UI in the background, e.g. scrolling the page.
The feature onboarding drawer closes

1. manually when pressing the close button in the drawer
2. automatically when navigating away from the page
3. automatically when doing af gestures, e.g. swipe, longpress etc. (but not scroll!)
4. automatically when rotating the device 

**Describe alternatives you've considered**
We have thought about prioritising the requirements but it was quite hard since its not estimated. Drawer width might be something that can be a "nice to have" the same goes for the close button, dependant how hard it is to change it.

**Additional context**

https://zpl.io/bopXE3M
Onboarding_drawer_examples pmg

Onboarding_drawer

### PR DESCRIPTION
**Pull request by lydemann**
This PR closes #1283

Feature hint

PR walkthrough: https://www.loom.com/share/2d9da535fff14bbe8775d41baab198f5

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #[replace with issue number]

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: libs/designsystem/src/lib/scss/_global-styles.scss [hunk 1]
```diff
@@ -46,6 +46,17 @@ ion-modal.kirby-overlay {
     }
   }
 
+  &.interact-with-background {
+    display: table;
```
Comment (lydemann) [inline libs/designsystem/src/lib/scss/_global-styles.scss:50]: This was the best way I found to make the parent fit child content

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss [hunk 2]
```diff
@@ -97,6 +97,7 @@ ion-title {
 
 ion-content {
   --background: transparent;
+  color: var(--kirby-modal-color);
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:100]: Please use [the public API](https://ionicframework.com/docs/api/content#css-custom-properties):
```suggestion
  --color: var(--kirby-modal-color);
```

Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:100]: Consider adding fallback:
```suggestion
  color: var(--kirby-modal-color, #{get-color('black')});
```

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss [hunk 3]
```diff
@@ -111,6 +112,10 @@ ion-content {
   --padding-end: #{size('s')};
 }
 
+button {
+  color: var(--kirby-modal-color);
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.scss:116]: Consider adding fallback:
```suggestion
  color: var(--kirby-modal-color, #{get-color('black')});
```

File: libs/designsystem/src/lib/scss/_global-styles.scss [hunk 4]
```diff
@@ -64,6 +75,12 @@ ion-modal.kirby-overlay {
     --border-radius: 0;
     --height: auto;
 
+    &.invert-colors {
+      --background: #{get-color('background-color-contrast')};
+      --kirby-modal-color: #{get-color('background-color')};
+      --ion-text-color: #{get-color('background-color')};
+    }
```
Comment (jakobe) [inline libs/designsystem/src/lib/scss/_global-styles.scss:82]: As per tech refinement, this should be handled by exposing css custom props for background + text color and let the target project set those

File: libs/designsystem/src/lib/components/modal/services/modal.helper.ts [hunk 5]
```diff
@@ -43,8 +44,11 @@ export class ModalHelper {
         config.flavor === 'drawer' ? 'kirby-drawer' : null,
         config.flavor === 'compact' ? 'kirby-modal-compact' : null,
         modalSize,
+        config.interactWithBackground ? 'interact-with-background' : null,
+        config.invertColors ? 'invert-colors' : null,
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/services/modal.helper.ts:48]: As per tech refinement, this should be handled by exposing css custom props for background + text color and let the target project set those

File: libs/designsystem/src/lib/scss/_global-styles.scss (hunk 1 above)
Comment (jakobe) [reply libs/designsystem/src/lib/scss/_global-styles.scss:50]: Let's talk about this ;)

File: apps/cookbook/src/app/examples/modal-example/modal-example-default.component.ts [hunk 6]
```diff
@@ -208,6 +220,8 @@ export class ModalExampleDefaultComponent {
     const config: ModalConfig = {
       flavor,
       component: FirstEmbeddedModalExampleComponent,
+      interactWithBackground: this.interactWithBackground,
+      invertColors: this.invertColors,
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-default.component.ts:224]: As per tech refinement, this should be handled by exposing css custom props for background + text color and let the target project set those

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts [hunk 7]
```diff
@@ -40,6 +40,12 @@ export class ModalExampleConfigurationComponent {
   @Input() openFullHeight: boolean;
   @Output() openFullHeightChange = new EventEmitter<boolean>();
 
+  @Input() interactWithBackground: boolean;
+  @Output() interactWithBackgroundChange = new EventEmitter<boolean>();
+
+  @Input() invertColors: boolean;
+  @Output() invertColorsChange = new EventEmitter<boolean>();
+
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts:48]: As per tech refinement, this should be handled by exposing css custom props for background + text color and let the target project set those

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.html [hunk 8]
```diff
@@ -57,3 +57,17 @@
...
+  (checkedChange)="toggleInteractWithBackground($event)"
+  text="Interact with background (drawer only)"
+></kirby-checkbox>
+
+<kirby-checkbox
+  *ngIf="invertColors !== undefined"
+  [checked]="invertColors"
+  (checkedChange)="toggleInvertColors($event)"
+  text="Invert colors (drawer only)"
+></kirby-checkbox>
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.html:73]: As per tech refinement, this should be handled by exposing css custom props for background + text color and let the target project set those

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/scss/_global-styles.scss (hunk 1 above)
Comment (lydemann) [reply libs/designsystem/src/lib/scss/_global-styles.scss]: Now the ion-modal is positioned to fit content

Comment (lydemann) [reply libs/designsystem/src/lib/scss/_global-styles.scss]: Using js

File: apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts [hunk 9]
```diff
@@ -67,6 +70,19 @@ export class ModalShowcaseComponent implements AfterViewInit {
...
+    {
+      name: 'interactWithBackground',
+      description: `(Optional) Removes backdrop and allows interaction with the background.
+      Please note: Only available on modals with a \`drawer\` flavor.`,
+      defaultValue: 'false',
+      type: ['boolean'],
+    },
+    {
+      name: 'cssClass',
+      description: `(Optional) Adds custom css classes to the modal. These needs to be styled with global styling.`,
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/modal-showcase/modal-showcase.component.ts:82]:
```suggestion
      description: `(Optional) Adds custom css classes to the modal. This allows for custom styling of the modal (see 'CSS Custom Properties' section).`,
```

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 10]
```diff
@@ -93,6 +99,7 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     }
     return this._intersectionObserver;
   }
+  private destroy = new Subject();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:102]:
```suggestion
  private destroy$ = new Subject();
```

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 11]
```diff
@@ -125,6 +132,26 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
...
+            const horizontalPosition =
+              (this.windowRef.document.documentElement.offsetWidth -
+                this.elementRef.nativeElement.scrollWidth) /
+              2;
+            this.ionModalElement.style.top = `${topPosition}px`;
+            this.ionModalElement.style.left = `${horizontalPosition}px`;
+            this.ionModalElement.style.right = `${horizontalPosition}px`;
+          });
+        });
+    }
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:154]: Consider readability and extract to separate method (same as `this.initializeSizing, this.initializeModalRoute` etc

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts [hunk 12]
```diff
@@ -125,6 +132,26 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
...
+        .subscribe(() => {
+          // wait for template to render
+          setTimeout(() => {
+            const topPosition =
+              this.windowRef.document.documentElement.offsetHeight -
+              this.elementRef.nativeElement.scrollHeight;
+            const horizontalPosition =
+              (this.windowRef.document.documentElement.offsetWidth -
+                this.elementRef.nativeElement.scrollWidth) /
+              2;
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts:148]: Consider readability:
```suggestion
            const docElement = this.windowRef.document.documentElement;
            const hostElement = this.elementRef.nativeElement;
            const topPosition = docElement.offsetHeight - hostElement.scrollHeight;
            const horizontalPosition = (docElement.offsetWidth - hostElement.scrollWidth) / 2;
```

File: libs/designsystem/src/lib/scss/_global-styles.scss [hunk 13]
```diff
@@ -46,6 +46,12 @@ ion-modal.kirby-overlay {
     }
   }
 
+  &.interact-with-background {
+    &.kirby-modal.kirby-drawer {
+      --kirby-modal-padding-top: 0px;
+    }
+  }
+
```
Comment (jakobe) [inline libs/designsystem/src/lib/scss/_global-styles.scss:54]: Consider moving to existing `&.kirby-drawer` section under `&.kirby-modal` section:
```
&.kirby-modal {
  ...
   &.kirby-drawer {
    ...
    &.interact-with-background {
      --kirby-modal-padding-top: 0px;
    }
  }
}
```

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts [hunk 14]
```diff
@@ -91,6 +91,50 @@ describe('ModalWrapperComponent', () => {
...
+
+      spectator.component['ionModalDidPresent'].next();
+      spectator.component['ionModalDidPresent'].complete();
+
+      await TestHelper.waitForTimeout();
+      spectator.detectChanges();
+
+      expect(spectator.component['ionModalElement'].style.top).toBe('330px');
+      expect(spectator.component['ionModalElement'].style.left).toBe('400px');
+      expect(spectator.component['ionModalElement'].style.right).toBe('400px');
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts:117]: Are these related to `TestHelper.screensize.desktop`? If so, consider readability and calculate/somehow convey the relation/expected values...

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts [hunk 15]
```diff
@@ -91,6 +91,50 @@ describe('ModalWrapperComponent', () => {
...
+
+      spectator.component['viewportResize'].next();
+      spectator.component['viewportResize'].complete();
+
+      await TestHelper.waitForResizeObserver();
+      spectator.detectChanges();
+
+      expect(spectator.component['ionModalElement'].style.top).toBe('330px');
+      expect(spectator.component['ionModalElement'].style.left).toBe('400px');
+      expect(spectator.component['ionModalElement'].style.right).toBe('400px');
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts:134]: Are these related to `TestHelper.screensize.desktop`? If so, consider readability and calculate/somehow convey the relation/expected values...

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts [hunk 16]
```diff
@@ -99,4 +105,14 @@ export class ModalExampleConfigurationComponent {
     this.openFullHeight = show;
     this.openFullHeightChange.emit(this.openFullHeight);
   }
+
+  toggleInteractWithBackground(show: boolean) {
+    this.interactWithBackground = show;
+    this.interactWithBackgroundChange.emit(show);
+  }
+
+  toggleInvertColors(show: boolean) {
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts:114]: Consider naming:
```suggestion
  toggleCustomCssClass(show: boolean) {
```

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts [hunk 17]
```diff
@@ -40,6 +40,12 @@ export class ModalExampleConfigurationComponent {
   @Input() openFullHeight: boolean;
   @Output() openFullHeightChange = new EventEmitter<boolean>();
 
+  @Input() interactWithBackground: boolean;
+  @Output() interactWithBackgroundChange = new EventEmitter<boolean>();
+
+  @Input() invertColors: boolean;
+  @Output() invertColorsChange = new EventEmitter<boolean>();
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.ts:47]: Consider naming:
```suggestion
  @Input() customCssClass: boolean;
  @Output() customCssClass Change = new EventEmitter<boolean>();
```

File: apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.html [hunk 18]
```diff
@@ -57,3 +57,17 @@
...
+  [checked]="interactWithBackground"
+  (checkedChange)="toggleInteractWithBackground($event)"
+  text="Interact with background (drawer only)"
+></kirby-checkbox>
+
+<kirby-checkbox
+  *ngIf="invertColors !== undefined"
+  [checked]="invertColors"
+  (checkedChange)="toggleInvertColors($event)"
+  text="Invert colors (drawer only)"
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/modal-example/modal-example-configuration/modal-example-configuration.component.html:72]: Consider wording:
```suggestion
  text="Add custom CSS class (drawer only)"
```

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts (hunk 14 above)
Comment (lydemann) [reply libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts]: It is not a testing best practice to duplicate calculations in the expected values. That is just code duplication.

**Review by jakobe — APPROVED** [review]
LGTM!

**Pull request merged**